### PR TITLE
flx(component): buttonColor is calculated in the 'var' case

### LINKED ISSUE #17315: [Component] [button] Button auto hover and active color not working when `color` obtained from css variable
Bug Type: **`Component`**

## Environment

- Vue Version: `3.3.4`
- Element Plus Version: `2.7.6`
- Browser / OS: `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36`
- Build Tool: `Vite`

### Related Component

- `el-button`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6IjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8aDE+VGVzdCBidXR0b24gY29sb3IgZnJvbSB2YXJpYWJsZTwvaDE+XG5cbiAgPGVsLWJ1dHRvbiBjb2xvcj1cInZhcigtLWN1c3RvbS1wcmltYXJ5LWNvbG9yKVwiPlRlc3QgQnV0dG9uPC9lbC1idXR0b24+XG48L3RlbXBsYXRlPlxuXG48c3R5bGU+XG46cm9vdHtcbiAgICAtLWN1c3RvbS1wcmltYXJ5LWNvbG9yOiAjNGU0NGE0O1xufVxuPC9zdHlsZT4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXSxcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJfbyI6e319)

### Steps to reproduce

```vue
<script setup lang="ts">
</script>

<template>
  <h1>Test button color from variable</h1>

  <el-button color="var(--custom-primary-color)">Test Button</el-button>
</template>

<style>
:root{
    --custom-primary-color: #4e44a4;
}
</style>
```

### What is Expected?

Given sets of `css` color variables, The `el-button` color of hover or active could auto calculated

### What is actually happening?

It can not auto inference corresponding hover and active color.
The actual hover color is default set to rgb(77,77,77), active color is rgb(4,4,4)

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by webvs2**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.
close #17315

### CODE REVIEW

**Comment by kooriookami** [comment]
Please fix the test

**Review by kooriookami — APPROVED** [review]

**Comment by webvs2** [comment]
> Please fix the test

😉

**Review by btea — APPROVED** [review]

File: packages/components/button/src/button-custom.ts [hunk 1]
```diff
@@ -17,9 +17,15 @@ export function useButtonCustomStyle(props: ButtonProps) {
   return computed(() => {
     let styles: Record<string, string> = {}
 
-    const buttonColor = props.color
+    let buttonColor = props.color
 
     if (buttonColor) {
+      const match = (buttonColor as string).match(/var\((.*?)\)/)
+      if (match) {
+        buttonColor = getComputedStyle(
```
Comment (btea) [inline packages/components/button/src/button-custom.ts:25]: Is it better not to omit window?

File: packages/components/button/src/button-custom.ts [hunk 2]
```diff
@@ -17,9 +17,15 @@ export function useButtonCustomStyle(props: ButtonProps) {
...
-    const buttonColor = props.color
+    let buttonColor = props.color
 
     if (buttonColor) {
+      const match = (buttonColor as string).match(/var\((.*?)\)/)
+      if (match) {
+        buttonColor = window
+          .getComputedStyle(window.document.documentElement)
+          .getPropertyValue(match[1])
+      }
```
Comment (btea) [inline packages/components/button/src/button-custom.ts:28]: Do we need to be compatible with SSR, such as adding `isClient` judgment. 🤔

Comment (webvs2) [reply packages/components/button/src/button-custom.ts:28]: 🤔Is this only available on the client side?

Comment (btea) [reply packages/components/button/src/button-custom.ts:28]: It doesn't seem necessary, so let's leave it like this for now.

**Review by btea — APPROVED** [review]

**Pull request merged**